runtime/compiler/makefile.ftl

```diff
@@ -87,6 +87,9 @@ export BUILD_CONFIG?=prod
 <#elseif uma.spec.id?starts_with("linux_arm_linaro")>
   export PLATFORM=arm-linux-gcc-cross
 </#if>
+<#if uma.spec.id?starts_with("osx_x86-64")>
+  export PLATFORM=amd64-osx-clang
```

Comment:
@jdekonin for JDK8, we need to set `export PLATFORM=amd64-osx-gcc`. Nazim is adding `amd64-osx-gcc.mk` via https://github.com/eclipse/openj9/pull/3331. 

fyi - @DanHeidinga @0xdaryl @nbhuiyan